src/shogun/lib/any.h

```diff
@@ -1222,9 +1223,12 @@ namespace shogun {
 
 	template <typename T>
 	inline void register_casts() {
-		if constexpr (std::is_base_of<CSGObject, typename std::remove_pointer<T>::type>::value)
+		using Derived = std::remove_pointer_t<T>;
+		if constexpr (std::is_base_of_v<CSGObject, Derived>)
 		{
 			Any::register_caster<T, CSGObject*>([] (T value) { return dynamic_cast<CSGObject*>(value); });
+			if constexpr (!std::is_same_v<nullptr_t, base_type<Derived>>)
```

Comment:
not the nicest way....

Reply:
You can also write a is_any_of_derived to make it more explicit if you think this isn’t nice